### PR TITLE
updated Settings and UpdateSettingsOptions structs

### PR DESCRIPTION
**Pull request by mickymiek**
Updated settings.go structs according to [Gitlab api docs](https://docs.gitlab.com/ce/api/settings.html#change-application.settings)

### CODE REVIEW

File: settings.go [hunk 1]
```diff
@@ -30,25 +30,39 @@ type SettingsService struct {
...
+	DefaultGroupVisibility            string            `json:"default_group_visibility"`
+	DomainWhitelist                   []string          `json:"domain_whitelist"`
+	DomainBlacklist_Enabled           bool              `json:"domain_blacklist_enabled"`
+	DomainBlacklist                   []string          `json:"domain_blacklist"`
+	UserOauthApplications             bool              `json:"user_oauth_applications"`
+	AfterSignOutPath                  string            `json:"after_sign_out_path"`
+	ContainerRegistryTokenExpireDelay int               `json:"container_registry_token_expire_delay"`
+	RepositoryStorages                []string          `json:"repository_storages"`
+	KodingEnabled                     bool              `json:"koding_enabled"`
+	KodingUrl                         string            `json:"koding_url"`
```
Comment (svanharmelen) [inline settings.go]: Please make this `KodingURL` so `golint` doesn't complain about the name.

File: settings.go [hunk 2]
```diff
@@ -30,25 +30,39 @@ type SettingsService struct {
...
+	PlantumlEnabled                   bool              `json:"plantuml_enabled"`
+	PlantumlUrl                       string            `json:"plantuml_url"`
```
Comment (svanharmelen) [inline settings.go]: Please make this `PlantumURL` so `golint` doesn't complain about the name.

File: settings.go [hunk 3]
```diff
@@ -79,22 +93,36 @@ func (s *SettingsService) GetSettings(options ...OptionFunc) (*Settings, *Respon
...
+	DomainBlacklist                   []string          `url:"domain_blacklist,omitempty" json:"domain_blacklist,omitempty"`
+	UserOauthApplications             *bool             `url:"user_oauth_applications,omitempty" json:"user_oauth_applications,omitempty"`
+	AfterSignOutPath                  *string           `url:"after_sign_out_path,omitempty" json:"after_sign_out_path,omitempty"`
+	ContainerRegistryTokenExpireDelay *int              `url:"container_registry_token_expire_delay,omitempty" json:"container_registry_token_expire_delay,omitempty"`
+	RepositoryStorages                []string          `url:"repository_storages,omitempty" json:"repository_storages,omitempty"`
+	KodingEnabled                     *bool             `url:"koding_enabled,omitempty" json:"koding_enabled,omitempty"`
+	KodingUrl                         *string           `url:"koding_url,omitempty" json:"koding_url,omitempty"`
+	PlantumlEnabled                   *bool             `url:"plantuml_enabled,omitempty" json:"plantuml_enabled,omitempty"`
+	PlantumlUrl                       *string           `url:"plantuml_url,omitempty" json:"plantuml_url,omitempty"`
```
Comment (svanharmelen) [inline settings.go]: Please make this `PlantumURL` so `golint` doesn't complain about the name.

File: settings.go [hunk 4]
```diff
@@ -79,22 +93,36 @@ func (s *SettingsService) GetSettings(options ...OptionFunc) (*Settings, *Respon
...
+	DomainWhitelist                   []string          `url:"domain_whitelist,omitempty" json:"domain_whitelist,omitempty"`
+	DomainBlacklist_Enabled           *bool             `url:"domain_blacklist_enabled,omitempty" json:"domain_blacklist_enabled,omitempty"`
+	DomainBlacklist                   []string          `url:"domain_blacklist,omitempty" json:"domain_blacklist,omitempty"`
+	UserOauthApplications             *bool             `url:"user_oauth_applications,omitempty" json:"user_oauth_applications,omitempty"`
+	AfterSignOutPath                  *string           `url:"after_sign_out_path,omitempty" json:"after_sign_out_path,omitempty"`
+	ContainerRegistryTokenExpireDelay *int              `url:"container_registry_token_expire_delay,omitempty" json:"container_registry_token_expire_delay,omitempty"`
+	RepositoryStorages                []string          `url:"repository_storages,omitempty" json:"repository_storages,omitempty"`
+	KodingEnabled                     *bool             `url:"koding_enabled,omitempty" json:"koding_enabled,omitempty"`
+	KodingUrl                         *string           `url:"koding_url,omitempty" json:"koding_url,omitempty"`
```
Comment (svanharmelen) [inline settings.go]: Please make this `KodingURL` so `golint` doesn't complain about the name.

**Comment by mickymiek** [comment]
On it! Thanks

File: settings.go [hunk 5]
```diff
@@ -79,22 +152,95 @@ func (s *SettingsService) GetSettings(options ...OptionFunc) (*Settings, *Respon
...
-	RestrictedVisibilityLevels []VisibilityValue `url:"restricted_visibility_levels,omitempty" json:"restricted_visibility_levels,omitempty"`
-	MaxAttachmentSize          *int              `url:"max_attachment_size,omitempty" json:"max_attachment_size,omitempty"`
-	SessionExpireDelay         *int              `url:"session_expire_delay,omitempty" json:"session_expire_delay,omitempty"`
-	DefaultProjectVisibility   *int              `url:"default_project_visibility,omitempty" json:"default_project_visibility,omitempty"`
-	DefaultSnippetVisibility   *int              `url:"default_snippet_visibility,omitempty" json:"default_snippet_visibility,omitempty"`
-	RestrictedSignupDomains    []string          `url:"restricted_signup_domains,omitempty" json:"restricted_signup_domains,omitempty"`
-	UserOauthApplications      *bool             `url:"user_oauth_applications,omitempty" json:"user_oauth_applications,omitempty"`
-	AfterSignOutPath           *string           `url:"after_sign_out_path,omitempty" json:"after_sign_out_path,omitempty"`
+	AdminNotificationEmail              string            `url:"admin_notification_email,omitempty" json:"admin_notification_email,omitempty"`
```
Comment (svanharmelen) [inline settings.go]: All the types (except for slices) should be pointers.

Comment (mickymiek) [reply settings.go]: My bad, fixed

**Comment by svanharmelen** [comment]
Thanks @mickymiek! LGTM!

**Pull request merged**